api/rest/data.def.js

```diff
@@ -32,11 +34,17 @@ module.exports = function(config) {
     var module = {};
 
     function SubmitDataOption(data) {
+        var isBinary = common.isBinary(data.body);
         this.pathname = common.buildPath(api.data.SEND, data.deviceId);
         this.token = data.userToken;
-        ConnectionOptions.call(this);
+        ConnectionOptions.call(this, isBinary);
         this.method = 'POST';
-        this.body =  JSON.stringify(data.body);
+        if ( isBinary ) {
```

Comment:
Make sure you run "grunt" before creating a PR to test for code consistency, formatting and ensure all code tests pass. This if..else.. doesn't conform to the formatting for the project.